src/libpriv/rpmostree-core.c

```diff
@@ -3772,20 +3772,21 @@ rpmostree_context_get_tmprootfs_dfd  (RpmOstreeContext *self)
   return self->tmprootfs_dfd;
 }
 
-static gboolean
-pkg_name_is_kernel (const char *name)
-{
-  const char *kernel_names[] = {"kernel", "kernel-core", "kernel-rt", "kernel-rt-core", NULL};
-  return g_strv_contains (kernel_names, name);
-}
-
-/* Determine if a txn element contains vmlinuz.  TODO: Check via provides?
+/* Determine if a txn element contains vmlinuz via provides.
  * There's also some hacks for this in libdnf.
  */
 static gboolean
 rpmte_is_kernel (rpmte te)
 {
-  return pkg_name_is_kernel (rpmteN (te));
+  rpmds provides = rpmdsInit (rpmteDS (te, RPMTAG_PROVIDENAME));
+  while (rpmdsNext (provides) >= 0)
+   {
+     const char *provname = rpmdsN (provides);
+     g_printerr ("checking provided %s for kernel\n", provname);
+     if (g_str_equal (provname, "installonlypkg(kernel)") || g_str_equal (provname, "kernel"))
```

Comment:
Hmm, I'm not sure if we should use `installonlypkg` for this. IIRC that mechanism is meant to solve a separate issue, which is the install of multiple versions of the same package.

Checking for `kernel` or `kernel-rt` in the provides list seems reasonable though.

Reply:
Agree.

Also at a more general level...I think it's just going to work better if kernel forks use the same *name* as the kernel package (and more generally, the spec file), and just differentiate via the version/release tags.

Reply:
Agreed. I'll make that change then since `installonlypkg` isn't well suited for this.

Also as you mentioned, wouldn't it be wise to also check if other kernel package names `{"kernel", "kernel-core", "kernel-rt", "kernel-rt-core"}` are provided?

Reply:
@cgwalters This should be resolved with the latest commit